rust/src/chain_crypto/algorithms/ed25519_extended.rs

```diff
@@ -0,0 +1,89 @@
+use crate::chain_crypto::key::{AsymmetricKey, AsymmetricPublicKey, SecretKeyError, SecretKeySizeStatic};
+use crate::chain_crypto::sign::SigningAlgorithm;
+
+use super::ed25519 as ei;
+
+use cryptoxide::ed25519;
+use rand_os::rand_core::{CryptoRng, RngCore};
+
+use ed25519_bip32::{XPrv, XPRV_SIZE};
+
+/// ED25519 Signing Algorithm with extended secret key
+pub struct Ed25519Extended;
+
+#[derive(Clone)]
+pub struct ExtendedPriv([u8; ed25519::PRIVATE_KEY_LENGTH]);
+
+impl AsRef<[u8]> for ExtendedPriv {
+    fn as_ref(&self) -> &[u8] {
+        &self.0[..]
+    }
+}
+
+impl ExtendedPriv {
+    pub fn from_xprv(xprv: &XPrv) -> Self {
+        let mut buf = [0; ed25519::PRIVATE_KEY_LENGTH];
+        xprv.get_extended_mut(&mut buf);
```

Comment:
Why did this get changed from `get_extended` to `get_extended_mut`?

It seems this change originates from this upstream commit (https://github.com/input-output-hk/chain-libs/commit/0295b6db5fea512838a957739cc33746c6d0da88), and the code is only partially ported into this PR. I'm not sure if it's safe to replace just this part and not make the rest of the code changes from this commit.